privacy_evaluator/attacks/membership_inference/black_box.py

```diff
@@ -27,46 +27,31 @@ def __init__(
         :param y_train: Labels for the data that was used to train the target model.
         :param x_test: Data that was not used to train the target model.
         :param y_test: Labels for the data that was not used to train the target model.
-        :param attack_train_ratio: Ratio between used test and train data from the target model to train the attack model.
-        """
-        super().__init__(target_model, x_train, y_train, x_test, y_test)
-
-        self.attack_train_ratio = attack_train_ratio
-        self.attack_train_size = int(len(x_train) * attack_train_ratio)
-        self.attack_test_size = int(len(x_test) * attack_train_ratio)
-
-    def infer(
-        self, attack_model_type: str = "nn", *args, **kwargs
-    ) -> Tuple[np.ndarray, np.ndarray]:
-        """Alias method for attack().
-
         :param attack_model_type: Type of the attack model. On of "rf", "gb", "nn".
-        :param args: Arguments of the attack.
-        :param kwargs: Keyword arguments of the attack.
-        :return: Two arrays holding the inferred membership status. The first array includes the results for the
-        inferred membership status of the train data and the second includes the results for the test data, where 1
-        indicates a member and 0 indicates non-member. The optimal attack would return only ones for the first array and
-        only zeros for the second.
+        :raises TypeError: If `attack_model_type` is of invalid type.
+        :raises ValueError: If `attack_model_type` is none of `rf`, `gb`, `nn`.
         """
-        assert attack_model_type in ["rf", "gb", "nn"]
-
-        attack = MembershipInferenceBlackBox(
-            self.target_model.art_classifier, attack_model_type=attack_model_type
+        if not isinstance(attack_model_type, str):
```

Comment:
It's good that you do type checking, but you hard-code the value above, if I see that right, so you would never run into these errors? A change above will do.

Reply:
As @jtorhoff mentioned above, the value is not hardcoded, it is just a default value. someone can still pass invalid values here.